src/priv_verif_token.ts

```diff
@@ -0,0 +1,273 @@
+            throw new Error('blinded message has invalide size');
+        }
+
+        this.tokenType = VOPRF.value;
+    }
+
+    static deserialize(bytes: Uint8Array): TokenRequest2 {
+        let offset = 0;
+        const input = new DataView(bytes.buffer);
+
+        const type = input.getUint16(offset);
+        offset += 2;
+
+        if (type !== VOPRF.value) {
+            throw new Error('mismatch of token type');
+        }
+
+        const truncatedTokenKeyId = input.getUint8(offset);
+        offset += 1;
+
+        const len = VOPRF.Ne;
+        const blindedMsg = new Uint8Array(input.buffer.slice(offset, offset + len));
+        offset += len;
+
+        return new TokenRequest2(truncatedTokenKeyId, blindedMsg);
+    }
+
+    serialize(): Uint8Array {
+        const output = new Array<ArrayBuffer>();
+
+        let b = new ArrayBuffer(2);
+        new DataView(b).setUint16(0, this.tokenType);
+        output.push(b);
+
+        b = new ArrayBuffer(1);
+        new DataView(b).setUint8(0, this.truncatedTokenKeyId);
+        output.push(b);
+
+        b = this.blindedMsg.buffer;
+        output.push(b);
+
+        return new Uint8Array(joinAll(output));
+    }
+}
+
+export class TokenResponse2 {
+    // struct {
+    //     uint8_t evaluate_msg[Ne];
+    //     uint8_t evaluate_proof[Ns+Ns];
+    //  } TokenResponse;
+
+    constructor(
+        public readonly evaluateMsg: Uint8Array,
+        public readonly evaluateProof: Uint8Array,
+    ) {
+        if (evaluateMsg.length !== VOPRF.Ne) {
+            throw new Error('evaluate_msg has invalid size');
+        }
+        if (evaluateProof.length !== 2 * VOPRF.Ns) {
+            throw new Error('evaluate_proof has invalid size');
+        }
+    }
+
+    static deserialize(bytes: Uint8Array): TokenResponse2 {
+        let offset = 0;
+        let len = VOPRF.Ne;
+        const evaluateMsg = new Uint8Array(bytes.slice(offset, offset + len));
+        offset += len;
+
+        len = 2 * VOPRF.Ns;
+        const evaluateProof = new Uint8Array(bytes.slice(offset, offset + len));
+
+        return new TokenResponse2(evaluateMsg, evaluateProof);
+    }
+
+    serialize(): Uint8Array {
+        return new Uint8Array(joinAll([this.evaluateMsg, this.evaluateProof]));
+    }
+}
+
+export function verifyToken2(token: Token, privateKeyIssuer: Uint8Array): Promise<boolean> {
+    const vServer = new VOPRFServer(VOPRF.suite, privateKeyIssuer);
+    const authInput = token.authInput.serialize();
+    return vServer.verifyFinalize(authInput, token.authenticator);
+}
+
+export class Issuer2 {
+    private vServer: VOPRFServer;
+
+    constructor(
+        public name: string,
+        private privateKey: Uint8Array,
+        public publicKey: Uint8Array,
+    ) {
+        this.vServer = new VOPRFServer(VOPRF.suite, this.privateKey);
+    }
+
+    async issue(tokReq: TokenRequest2): Promise<TokenResponse2> {
+        const blindedElt = VOPRF.group.desElt(tokReq.blindedMsg);
```

Comment:
```typescript
export class EvaluationRequest {
    constructor(public readonly blinded: Array<Elt>) {}

    serialize(): Uint8Array {
        return joinAll(toU16LenPrefixClass(this.blinded))
    }
```

I guess the privacy pass protocol is "set" here, but it's odd not using Evaluation.deserialize and EvaluationRequest.deserialize?

Reply:
Oh, I see, so Evaluation in voprf-ts is of the Batched variety, from: 
https://datatracker.ietf.org/doc/draft-ietf-privacypass-batched-tokens/

Reply:
I can't really see any "canonical" encodings of OPRF types here: 
https://github.com/cloudflare/circl/blob/main/oprf/client.go

?  

From OPRF spec:
```
Applications serialize protocol messages between client and server
for transmission.  Elements and scalars are serialized to byte
arrays, and values of type Proof are serialized as the concatenation
of two serialized scalars.  Deserializing these values can fail, in
which case the application MUST abort the protocol raising a
DeserializeError failure.
```

I guess serialize is for matching the vectors? 

```
expect(toHexListClass(finData.blinds)).toEqual(vi.Blind)
expect(toHexListClass(evalReq.blinded)).toEqual(vi.BlindedElement)

const ev = await server.blindEvaluate(evalReq, info)
expect(toHexListClass(ev.evaluated)).toEqual(vi.EvaluationElement)
expect(ev.proof && toHexListClass([ev.proof])).toEqual(
    vi.Proof && vi.Proof.proof
)

const output = await client.finalize(finData, ev, info)
expect(toHexListUint8Array(output)).toEqual(vi.Output)
```

No, it seems everything there serialized is either Scalar/Element or Proof(Scalar*2).